stroyprombeton/models.py

```diff
@@ -41,13 +41,105 @@ def get_absolute_url(self):
+    mark = models.CharField(default='', max_length=500, blank=True, verbose_name=_('mark'))
+    specification = models.TextField(default='', blank=True, verbose_name=_('specification'),)
+    length = models.IntegerField(null=True, blank=True, verbose_name=_('length'))
+    width = models.IntegerField(null=True, blank=True, verbose_name=_('width'))
+    height = models.IntegerField(null=True, blank=True, verbose_name=_('height'))
+    weight = models.FloatField(null=True, blank=True, verbose_name=_('weight'))
+    volume = models.FloatField(null=True, blank=True, verbose_name=_('volume'))
+    diameter_out = models.IntegerField(null=True, blank=True, verbose_name=_('diameter out'))
+    diameter_in = models.IntegerField(null=True, blank=True, verbose_name=_('diameter in'))
+
+    is_new_price = models.NullBooleanField(
+        blank=True,
+        null=True,
+        verbose_name=_('is new price'),
+    )
+    price = models.FloatField(
+        blank=True,
+        default=0,
+        db_index=True,
+        verbose_name=_('price'),
+    )
+    in_stock = models.PositiveIntegerField(
+        default=0,
+        db_index=True,
+        verbose_name=_('in stock'),
+    )
+    is_popular = models.BooleanField(
+        default=False,
+        db_index=True,
+        verbose_name=_('is popular'),
+    )
+
+    def __str__(self):
+        return ' '.join(self.product.name, self.mark)
+
+
+# not inherited from `catalog_models.AbstractProduct`, because
+# AbstractProduct's set of fields is shared between Product and Option models.
+class Product(catalog_models.AdminTreeDisplayMixin, page_models.PageMixin):
+    objects = catalog_models.ProductManager()
+
+    name = models.CharField(max_length=255, db_index=True, verbose_name=_('name'))
     category = models.ForeignKey(
         Category,
         on_delete=models.CASCADE,
         related_name='products',
         verbose_name=_('category'),
     )
+    options = models.ForeignKey(
+        Option,
+        on_delete=models.CASCADE,
+        null=True, blank=True,
+        related_name='product',
+        verbose_name=_('option'),
+    )
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def url(self):
+        return self.get_absolute_url()
+
+    @property
+    def parent(self):
+        return self.category or None
+
+    def get_absolute_url(self):
+        return reverse('product', args=(self.id,))
+
+    def get_admin_tree_title(self):
+        return f'[{self.id}] {self.name} {self.mark}]'
+
+    def get_root_category(self):
+        return self.category.get_root()
+
+    def get_siblings(self, offset):
+        return (
+            self.__class__.objects
+            .active()
+            .filter(category=self.category)
+            .exclude(id=self.id)
+            .prefetch_related('category')
+            .select_related('page')[:offset]
+        )
+
+    # we'll remove this fields from Product model
```

Comment:
Let's create a todo, which will migrate fields from `Product` into `Option` and remove these by the way

Reply:
@artemiy312 , this huge set of subtasks for this task will be solve this issue.
I'll update the comment